test/Microsoft.NET.Pack.Tests/GivenThatThereAreImplicitPackageReferences.cs

```diff
@@ -0,0 +1,246 @@
+        public void Packing_a_netcoreapp_2_0_library_does_not_include_the_implicit_dependency()
+        {
+            if (UsingFullFrameworkMSBuild)
+            {
+                //  Disabled on full framework MSBuild until CI machines have VS with bundled .NET Core / .NET Standard versions
+                //  See https://github.com/dotnet/sdk/issues/1077
+                return;
+            }
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "PackNetCoreApp20Library",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = false
+            };
+
+            var dependencies = PackAndGetDependencies(testProject);
+
+            dependencies.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void Packing_a_netcoreapp_1_1_app_includes_the_implicit_dependency()
+        {
+            TestProject testProject = new TestProject()
+            {
+                Name = "PackNetCoreApp11App",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp1.1",
+                IsExe = true
+            };
+
+            var dependencies = PackAndGetDependencies(testProject);
+
+            dependencies.Count().Should().Be(1);
+            dependencies.Single().Attribute("id").Value
+                .Should().Be("Microsoft.NETCore.App");
+
+            //  Don't check the exact version so that the test doesn't break if we roll forward to new patch versions of the package
+            dependencies.Single().Attribute("version").Value
+                .Should().StartWith("1.1.");
+        }
+
+        [Fact]
+        public void Packing_a_netcoreapp_2_0_app_includes_the_implicit_dependency()
+        {
+            if (UsingFullFrameworkMSBuild)
+            {
+                //  Disabled on full framework MSBuild until CI machines have VS with bundled .NET Core / .NET Standard versions
+                //  See https://github.com/dotnet/sdk/issues/1077
+                return;
+            }
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "PackNetCoreApp20App",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true
+            };
+
+            var dependencies = PackAndGetDependencies(testProject);
+
+            dependencies.Count().Should().Be(1);
+            dependencies.Single().Attribute("id").Value
+                .Should().Be("Microsoft.NETCore.App");
+
+            //  Don't check the exact version so that the test doesn't break if we roll forward to new patch versions of the package
+            dependencies.Single().Attribute("version").Value
+                .Should().StartWith("2.0.");
+        }
+
+        [Fact]
+        public void Packing_a_multitargeted_library_includes_implicit_dependencies_when_appropriate()
+        {
+            if (UsingFullFrameworkMSBuild)
```

Comment:
There is a CoreMSBuildOnlyFactAttribute that you can use in all these cases. And there is a CoreMSBuildAndWindowsOnlyFactAttribute that you can use as well. Let's stick to this new approach now. Fix it here and in the other places you are adding as well.